lib/modules/manager/hermit/artifacts.spec.ts

```diff
@@ -0,0 +1,239 @@
+import type { StatusResult } from 'simple-git/promise';
+import { mockedFunction } from '../../../../test/util';
+import { GlobalConfig } from '../../../config/global';
+import { exec } from '../../../util/exec';
+import {
+  localPathIsSymbolicLink,
+  readLocalFile,
+  readLocalSymlink,
+} from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact } from '../types';
+import { updateArtifacts } from '.';
+
+jest.mock('../../../util/exec');
+jest.mock('../../../util/git');
+jest.mock('../../../util/fs');
+
+const execMock = mockedFunction(exec);
+const getRepoStatusMock = mockedFunction(getRepoStatus);
+
+const lstatsMock = mockedFunction(localPathIsSymbolicLink);
+const readlinkMock = mockedFunction(readLocalSymlink);
+const readFileMock = mockedFunction(readLocalFile);
+
+describe('modules/manager/hermit/artifacts', () => {
+  beforeEach(() => {
+    execMock.mockClear();
+    getRepoStatusMock.mockClear();
+  });
+
+  describe('updateArtifacts', () => {
+    it('should run hermit install for packages and return updated files', async () => {
+      lstatsMock.mockResolvedValue(true);
+
+      readlinkMock.mockResolvedValue('hermit');
+      readFileMock.mockResolvedValue('hermit');
+      GlobalConfig.set({ localDir: '' });
+
+      execMock.mockResolvedValue({
+        stdout: '',
+        stderr: '',
+      });
+      getRepoStatusMock.mockResolvedValue({
+        not_added: ['bin/go-1.17.1'],
+        deleted: ['bin/go-1.17'],
+        modified: ['bin/go', 'bin/jq'],
+        created: ['bin/jq-extra'],
+        renamed: [
+          {
+            from: 'bin/jq-1.5',
+            to: 'bin/jq-1.6',
+          },
+        ],
+      } as StatusResult);
```

Comment:
better use our `partial<T>()` function instead of `as` cast, as it hides future type issues